docs/dev/version-migration/migration-guide.md

```diff
@@ -0,0 +1,22 @@
+# `partiql-lang-kotlin` Version Migration Guide
+
+(Note: this folder's documents will likely be converted to a set of wikis on https://github.com/partiql/partiql-lang-kotlin/wiki
+with each major version bump on a separate wiki page.
+TBD if this document will remain as a markdown file on GitHub)
+
+The `partiql-lang-kotlin` [release notes](https://github.com/partiql/partiql-lang-kotlin/releases) list changes 
+between minor versions as well as the last minor version to the subsequent major version (e.g. v0.2.7 -> v0.3.0). This 
+document lists the aggregated release notes between major `partiql-lang-kotlin` versions (e.g. v0.2.7 -> v0.3.4). To 
+make it easier to understand differences between `partiql-lang-kotlin` major versions, we've also
+* unified the change format (new features, breaking changes - behavioral and API, deprecated items, misc/bug fixes)
+* cleaned up some release note items (summarized sequences of related commits, omitted commits related to tests and 
+build-related changes)
+* separated CLI/REPL changes to a separate section
+* found other breaking changes from the compatibility reports between versions using the [japi-compliance-checker](https://github.com/lvc/japi-compliance-checker)
+
+The repo, [test-partiql-version-migration](https://github.com/alancai98/test-partiql-version-migration), hosts all the 
```

Comment:
Code migration examples are in Kotlin currently. Also need to gauge whether adding Java examples would be helpful.

Reply:
Personally I would not include Java migration examples because it may set a precedent to include both Kotlin and Java examples moving forward. Java/Kotlin back-and-forth is simple enough, that we might as well keep these clean and minimal. It would be nice to have something like Gradles examples with the groovy/kotlin tabs, but since these are just markdown docs, its simple enough to just do Kotlin.

Reply:
I'm also leaning towards not including Java migration examples since moving between Java and Kotlin is pretty straightforward imo. Could consider it a future task if users really want to see these examples in Java.